packages/core-components/src/components/DependencyGraph/Edge.tsx

```diff
@@ -80,7 +80,7 @@ const createPath = d3Shape
   .line<EdgePoint>()
   .x(d => d.x)
   .y(d => d.y)
-  .curve(d3Shape.curveMonotoneX);
```

Comment:
To follow up on @awanlin's comment, thinking that we should make it configurable like we have done the other properties like `Direction` and `Alignment` over here https://github.com/backstage/backstage/blob/2a0ad49da095cbc471ca54e0e8a05d9ad66e7b65/packages/core-components/src/components/DependencyGraph/types.ts#L83-L100

Maybe something like `Curve` or something which will translate the enum into a `d3shape` without exporting the `d3` types in the `api-report`

Reply:
@benjdlambert Thanks for your comment!
Do you have any suggestions on how this shoul be implemented ? 
> something will translate the enum into a d3shape without exporting the d3 types in the api-report

Reply:
So I'd say have a prop which defaults to `curveMonotoneX` but you can say `curve="stepBefore"` or something in the component to switch out the curve setting in D3 and also the changes in the svg if that works?

Reply:
Also sorry for the late reply, this totally shot under my radar.

Reply:
Yeah, so you would export a type which is `type Curve = 'stepBefore' | 'curveMonotoneX'` and this would remove the need to leek any of the d3 types in the api report. Internally we would check what the input is and map that to the d3 option.

Reply:
Thanks @benjdlambert! I have just pushed some changes